### PR TITLE
Add support for other virtual environment paths

### PR DESCRIPTION
**Pull request by lazyguru**
This maybe solves #3

At the moment, this does 2 things:
1) Adds a new configuration setting for venv_locations and defaults it to venv, env
2) If a virtual environment is not found and not able to be activated, it will refuse to start the site

Obviously, 2 is a bit opinionated as it makes the assumption that _everyone_ is using virtual environments.  It would be pretty easy to provide a hack for this if someone doesn't want to use virtual environments, but even then IMO they *should* be using virtual environments.

NOTE: I haven't tested this one at all yet. Just creating a draft PR so you can code review and let me know what you think about the code and 2 from above.

### CODE REVIEW

File: src/foreman/Command.py [hunk 1]
```diff
@@ -11,11 +11,14 @@ def startup(self):
             site = directory.split('/')[-2]
             activation_environment = self.find_virtual_environment_activation_file(directory)
 
-            command = f"cd {directory}"
-            command += f" && source {activation_environment}"
-            command += f" && pip install uwsgi && set -m; nohup uwsgi --socket /tmp/{site}.test.sock --wsgi-file wsgi.py &> /dev/null &"
-            subprocess.run(command, shell=True, close_fds=True,
-                           env={'PYTHONPATH': f'{directory}'})
+            if activation_environment is None:
+                self.warn(f"No virtual environment detected, not starting site {site}")
```
Comment (josephmancuso) [inline src/foreman/Command.py]: I recommend you don't just tell people problems. They know their site didn't start. Try telling them solutions like "No virtual environment detected, not starting site {site}. Please register your venv by running `foreman register /path/to/venv`"

Comment (lazyguru) [reply src/foreman/Command.py]: Fixed

**Comment by lazyguru** [comment]
@josephmancuso any thoughts on this PR?

**Comment by josephmancuso** [comment]
Don't we have a way to register the activated virtual environment location?

**Comment by lazyguru** [comment]
> Don't we have a way to register the activated virtual environment location?

Yes, use `foreman register <path/to/venv>`. This is called out in the error message displayed to the user:

```
self.line(f"<error>No virtual environment detected, not starting site {site}.</error>")
self.line("<error>Please register your venv by running:</error>")
self.line("<fg=magenta;options=bold>    foreman register /path/to/venv</>")
```

EDIT: Just realized you were talking about using the currently activated environment.  I can take a look at that. Do you think we should hold up the current PR for that?

**Comment by lazyguru** [comment]
Quick Google search provides this: https://stackoverflow.com/a/1883251/564576

**Comment by lazyguru** [comment]
@josephmancuso ok, I added support for already running inside a virtual environment. I also cleaned up the driver check to just display an error (this works nicer when you have multiple sites and only one of them is missing a driver). Take a look and let me know what you think

**Pull request merged**